File: kwAlgorithmClass/완전탐색/최소직사각형.py
def solution(sizes):
    # 해결 못함
    # answer = set()
    # for w, h in sizes:
    #     answer.add((h, w))
    #     answer.add((w, h))
    #
    # print(answer)
    # minH = 1000
    # minW = 1000
    # answer = list(answer)
    #
    # for w, h in answer:
    #     if minH * minW > w * h:
    #         minH = h
    #         minW = w
    #
    # return minW * minH

    # 쌤 풀이
    a = [max(i) for i in sizes]  # 모든 쌍 중 최대 값만
    b = [min(i) for i in sizes]  # 모든 쌍 중 최소 값만
    # a, b에는 모든 쌍들이 각각 1번씩만 들어 가게 된다. 쌍은 작거나 크거나 둘 증 하나 이므로
    return max(a) * max(b)  # 최댓값 중에 최댓값 * 최솟값 중에 최댓값

File: JAVA/src/최소_직사각형.java
public class 최소_직사각형 {

    class Solution {
        public int solution(int[][] sizes) {
            int answer = 0;
            int maxC=0, maxR=0;
            for(int i=0; i<sizes.length; i++){
                if(sizes[i][1] < sizes[i][0]){
                    int temp = sizes[i][0];
                    sizes[i][0] = sizes[i][1];
                    sizes[i][1] = temp;
                }
                maxC = maxC < sizes[i][0] ? sizes[i][0] : maxC;
                maxR = maxR < sizes[i][1] ? sizes[i][1] : maxR;
            }
            System.out.printf("%d %d", maxC, maxR);
            return maxC*maxR;
        }
    }
}
